src/Battleships/GameInterface.java

```diff
@@ -46,29 +65,153 @@ private char[][] emptygrid(char[][] gameGrid) {
+                ui.print(i + " ");
+                continue;
+            }
+            ui.print(i);
+        }
+        ui.print("\n");
+        for (int i = 0; i < grid.length; i++) {
+            ui.print(i);
+            for (int j = 0; j < grid.length; j++) {
+                ui.print(grid[i][j]);
+            }
+            ui.print("\n");
+        }
+    }
+
+    private boolean isIllegal(PlayerMovement playerMovement) {
+        int currentPlaya = battleships.getCurrentGameState().getCurrentPlayer();
+        try {
+            rulesChecker.isValidMove(battleships, playerMovement, currentPlaya);
+
+        } catch (IllegalArgumentException e) {
+            ui.print(e.getMessage() + ", type in your move again. \n");
+            return true;
+        } catch (ArrayIndexOutOfBoundsException e) {
+            ui.print(e.getMessage() + ", type in your move again. \n");
+            return true;
+        }
+        return false;
+    }
+
+
+    private boolean isIllegalShot(PlayerMovement playerMovement) {
+//        start here, implement code that will switch players grids between each other
+        int currentPlaya = battleships.getCurrentGameState().getCurrentPlayer();
+        try {
+            rulesChecker.isValidShot(battleships, playerMovement, currentPlaya);
+
+        } catch (IllegalArgumentException e) {
+            ui.print(e.getMessage() + ", type in your move again. \n");
+            return true;
+        } catch (ArrayIndexOutOfBoundsException e) {
+            ui.print(e.getMessage() + ", type in your move again. \n");
+            return true;
+        }
+        return false;
+    }
+
+    private void playersSwitcher(int player) {
+        if (player == 1) {
+            battleships.getCurrentGameState().setCurrentPlayer(2);
+        }
+        if (player == 2) {
+            battleships.getCurrentGameState().setCurrentPlayer(1);
+        }
+    }
+
+    private void optionAfterEachMove() throws InterruptedException {
+        ui.println("What would you like to do next?");
+        ui.println("Display my ships. Press 1.");
+        ui.println("Display my given shoots so far. Press 2.");
+        ui.println("End my turn. Press 3.");
+        String choice = ui.nextLine();
+        if (choice.matches("1")) {
+            int currentPlayer = battleships.getCurrentGameState().getCurrentPlayer();
+            printGrid(battleships.getCurrentGameState().getPlayer(currentPlayer - 1).getGameGrid());
+            optionAfterEachMove();
+        }
+        if (choice.matches("2")) {
+            int currentPlayer = battleships.getCurrentGameState().getCurrentPlayer();
+            printGrid(battleships.getCurrentGameState().getPlayer(currentPlayer - 1).getGivenShots());
+            optionAfterEachMove();
+        }
+        if (choice.matches("3")) {
+            for (int i = 5; i > 0; i--) {
+                ui.println("Your turn will end for " + i + " seconds. Move away from monitor.");
+                Thread.sleep(1000);
```

Comment:
Catch and handle it here or catch it and rethrow it as a custom RuntimeException, that way you won't have to throws InterruptedException in execution tree (I mean all the other methods that you had to add throws InterruptedException to).